chain/xrp/client/client.go

```diff
@@ -0,0 +1,718 @@
+}
+
+type SubmitResultTxJson struct {
+	Account            string   `json:"Account"`
+	Amount             *Balance `json:"Amount"`
+	Destination        string   `json:"Destination"`
+	Fee                string   `json:"Fee"`
+	Flags              int64    `json:"Flags"`
+	LastLedgerSequence int64    `json:"LastLedgerSequence"`
+	Sequence           int64    `json:"Sequence"`
+	SigningPubKey      string   `json:"SigningPubKey"`
+	TransactionType    string   `json:"TransactionType"`
+	TxnSignature       string   `json:"TxnSignature"`
+	Hash               string   `json:"hash"`
+}
+
+type LedgerResponse struct {
+	Result LedgerResult `json:"result"`
+}
+
+type LedgerResult struct {
+	Ledger             LedgerInfo `json:"ledger"`
+	LedgerCurrentIndex int64      `json:"ledger_current_index"`
+	Validated          bool       `json:"validated"`
+	Status             string     `json:"status"`
+}
+
+type LedgerInfo struct {
+	Closed      bool   `json:"closed"`
+	LedgerIndex string `json:"ledger_index"`
+	ParentHash  string `json:"parent_hash"`
+}
+
+type TransactionResponse struct {
+	Result TransactionResult `json:"result"`
+}
+
+type TransactionResult struct {
+	Account            string          `json:"Account"`
+	Amount             string          `json:"Amount,omitempty"`
+	Destination        string          `json:"Destination,omitempty"`
+	Fee                string          `json:"Fee"`
+	Flags              int64           `json:"Flags"`
+	LastLedgerSequence int64           `json:"LastLedgerSequence"`
+	Sequence           int64           `json:"Sequence"`
+	SigningPubKey      string          `json:"SigningPubKey"`
+	TransactionType    string          `json:"TransactionType"`
+	TxnSignature       string          `json:"TxnSignature"`
+	Hash               string          `json:"hash"`
+	DeliverMax         string          `json:"DeliverMax,omitempty"`
+	TakerGets          TakerGets       `json:"TakerGets,omitempty"`
+	TakerPays          TakerPays       `json:"TakerPays,omitempty"`
+	CtID               string          `json:"ctid,omitempty"`
+	Meta               TransactionMeta `json:"meta"`
+	Validated          bool            `json:"validated"`
+	Date               int64           `json:"date"`
+	LedgerIndex        int64           `json:"ledger_index"`
+	InLedger           int64           `json:"inLedger"`
+	Status             string          `json:"status"`
+}
+
+type TakerGets struct {
+	StringValue string  `json:"-"`
+	AmountValue *Amount `json:"-"`
+	IsString    bool    `json:"-"`
+}
+
+// UnmarshalJSON is the custom unmarshal method for TakerGets
+func (tg *TakerGets) UnmarshalJSON(data []byte) error {
+	var str string
+	if err := json.Unmarshal(data, &str); err == nil {
+		tg.StringValue = str
+		tg.IsString = true
+		return nil
+	}
+
+	var amount Amount
+	if err := json.Unmarshal(data, &amount); err == nil {
+		tg.AmountValue = &amount
+		tg.IsString = false
+		return nil
+	}
+
+	return fmt.Errorf("TakerGets is neither a string nor an Amount")
+}
```

Comment:
this is the same as TakerPays.  Perhaps delete and just rename to `TakeGetsOrPays`